components/semaphore.rst

```diff
@@ -0,0 +1,79 @@
+.. index::
+   single: Semaphore
+   single: Components; Semaphore
+
+The Semaphore Component
+=======================
+
+    The Semaphore Component manages `semaphores`_, a mechanism to provide
+    exclusive access to a shared resource.
+
+.. versionadded:: 5.2
+
+    The Semaphore Component was introduced in Symfony 5.2.
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/semaphore
+
+.. include:: /components/require_autoload.rst.inc
+
+Usage
+-----
+
+Semaphore are used to guarantee exclusive access to some shared resource.
```

Comment:
I think it's important in this first paragraph to indicate the difference between the Lock and Semaphore component (as this is a frequent question). Why should I use which component? (maybe with some real use-cases?)

Reply:
Added a brief mention to that while merging. Thanks.